### PR TITLE
Bugfix for using basis transcoding since paged atlas feature

### PR DESCRIPTION
**Pull request by Jhonnyg**
Fixed an issue with using basis compression together with cubemap generation, where the output of the texture encoding produced an empty result. To fix this issue, we have separated the transcoding into two passes - a preprocessing pass for memory allocation of the result, and one to do the actual transcoding.

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [x] Add comments where needed
* [-] Documentation
	* [-] Make sure that API documentation is updated in code comments
	* [-] Make sure that manuals are updated (in github.com/defold/doc)
* [x] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [-] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [-] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [-] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [-] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/CubemapBuilder.java [hunk 1]
```diff
@@ -88,6 +88,9 @@ public void build(Task<Void> task) throws CompileExceptionError, IOException {
                 //
                 // So for cube map textures we don't flip on any axis, meaning the texture data begin at the
                 // upper left corner of the input image.
+
+
+                // NOTE: Setting the same input for more than one side will cause a NPE when generating!
```
Comment (Jhonnyg) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/CubemapBuilder.java:93]: This is an old bug it seems..

File: engine/graphics/src/vulkan/graphics_vulkan.cpp [hunk 2]
```diff
@@ -3177,6 +3177,19 @@ namespace dmGraphics
         //        per mipmap instead of batching in one cmd
         if (useStageBuffer)
         {
+            uint32_t slice_size = texDataSize / layer_count;
+
+        #ifdef __MACH__
+            // Note: There is an annoying validation issue on osx for layered compressed data that causes a validation error
+            //       due to misalignment of the data when using a stage buffer. The offsets in the stage buffer needs to be
+            //       8 byte aligned but for compressed data that is not the case for the lowest mipmaps.
```
Comment (JCash) [inline engine/graphics/src/vulkan/graphics_vulkan.cpp:3185]: So basically aligning the mipmap data on a 8 or 16 byte alignment when writing, and allocating an aligned memory block when reading?

File: engine/graphics/src/vulkan/graphics_vulkan.cpp [hunk 3]
```diff
@@ -3208,9 +3222,7 @@ namespace dmGraphics
                 params.m_MipMap, layer_count);
             CHECK_VK_ERROR(res);
 
-            uint32_t slice_size = texDataSize / layer_count;
-
-            VkBufferImageCopy vk_copy_regions[6];
+            VkBufferImageCopy* vk_copy_regions = new VkBufferImageCopy[layer_count];
```
Comment (JCash) [inline engine/graphics/src/vulkan/graphics_vulkan.cpp:3225]: What is the max layer count?
If we can avoid dynamic mem allocations with little impact, I think we prefer that.

**Review by JCash — DISMISSED** [review]

File: engine/graphics/src/vulkan/graphics_vulkan.cpp (hunk 2 above)
Comment (Jhonnyg) [reply engine/graphics/src/vulkan/graphics_vulkan.cpp:3185]: Yes, I tried that and it didn't look correct. I'll have to revisit it.. I'll make an issue about it

**Pull request merged**